samples/99_zil_interpreter/app/builtins.rb

```diff
@@ -0,0 +1,21 @@
+class FunctionError < StandardError; end
+
+def define_for_evaled_args(&implementation)
+  lambda { |args, context|
+    evaled_args = args.map { |arg| eval_zil arg, context }
+    implementation.call(evaled_args, context)
+  }
+end
+
+ZIL_BUILTINS = {}
+
+ZIL_BUILTINS[:LVAL] = define_for_evaled_args { |args, context|
+  var_atom = args[0]
+  raise FunctionError, "No local value for #{var_atom.inspect}" unless context.locals.key? var_atom
+
+  context.locals[var_atom]
+}
+
+ZIL_BUILTINS[:+] = define_for_evaled_args { |args|
+  args.inject(0, :+)
+}
```

Comment:
Assuming I'm not reading something wrong here, `args` (in this file) is no longer our typical DRGTK `args` that we know and love.  Given the significance of `args` in DRGTK perhaps this should be called `arglist` to differentiate?

Reply:
good point... I will change that for clarity!

Reply:
Fixed it in 0b1308a - I chose `arguments` as a name

Also added a proper exception when a function is not found